### PR TITLE
TIP-538: delete the FilePathResolver

### PR DESCRIPTION
**Pull request by jjanvier**
The `FilePathResolver` service was injected in all export file writers. This really simple service was used to just create an option resolver.

Instead  of injecting it everywhere, the option resolver have been created in the `AbstractFileWriter`.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Changelog updated                 | will be updated with the whole import/export thing
| Review and 2 GTM                  | pleeeeaaaaase
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

**Comment by skeleton** [comment]
gtm

**Comment by jjanvier** [comment]
2 behat fail, but they also fail on master
(unstable according to @pierallard <3 )

**Comment by nidup** [comment]
Ok, gtm

**Comment by phaseinducer** [comment]
GTM, much simpler :+1:

File: src/Pim/Component/Connector/Writer/File/Csv/ProductWriter.php [hunk 1]
```diff
@@ -37,18 +36,16 @@ class ProductWriter extends AbstractFileWriter implements ItemWriterInterface, A
     protected $writtenFiles = [];
 
     /**
-     * @param FilePathResolverInterface $filePathResolver
      * @param BufferFactory             $bufferFactory
      * @param BulkFileExporter          $mediaCopier
      * @param FlatItemBufferFlusher     $flusher
```
Comment (pierallard) [inline src/Pim/Component/Connector/Writer/File/Csv/ProductWriter.php]: For your presentation JJ :trollface: 

"Please align"

**Comment by pierallard** [comment]
GTM

**Pull request merged**